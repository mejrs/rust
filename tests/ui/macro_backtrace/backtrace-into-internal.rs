//! We try to hide the internals of builtin-like macros from error messages,
//! but *not* if `-Z macro-backtrace` is enabled

//@ revisions: default with
//@[with] compile-flags: -Z macro-backtrace
//@[with] error-pattern: in this expansion of `macro_two!`

#![feature(allow_internal_unstable)]

#[allow_internal_unstable(hint_must_use, liballoc_internals)]
macro_rules! macro_one {
    ($($arg:tt)*) => {
        std::convert::identity(macro_two!($($arg)*))
    }
}

#[allow_internal_unstable(fmt_internals)]
macro_rules! macro_two {
    ($($arg:tt)*) => {
        { $($arg)* + 1 }
        //~^ERROR cannot add `{integer}` to `&str`
    }
}

fn main(){
    let _ = macro_one!("boo");
}
