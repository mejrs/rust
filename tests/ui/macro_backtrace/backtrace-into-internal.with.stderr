error[E0369]: cannot add `{integer}` to `&str`
  --> $DIR/backtrace-into-internal.rs:20:20
   |
LL | / macro_rules! macro_one {
LL | |     ($($arg:tt)*) => {
LL | |         std::convert::identity(macro_two!($($arg)*))
   | |                                -------------------- in this macro invocation (#2)
LL | |     }
LL | | }
   | |_- in this expansion of `macro_one!` (#1)
...
LL | / macro_rules! macro_two {
LL | |     ($($arg:tt)*) => {
LL | |         { $($arg)* + 1 }
   | |                    ^ - {integer}
...  |
LL | | }
   | |_- in this expansion of `macro_two!` (#2)
...
LL |       let _ = macro_one!("boo");
   |               -----------------
   |               |          |
   |               |          &str
   |               in this macro invocation (#1)

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0369`.
